pkg/networkservice/chains/nsmgr/server_test.go

```diff
@@ -45,66 +45,30 @@ import (
 	"github.com/networkservicemesh/sdk/pkg/tools/sandbox"
 )
 
-func TestNSMGR_RemoteUsecase(t *testing.T) {
+func TestNSMGR_OneHop(t *testing.T) {
+	t.Run("Local", func(t *testing.T) { testOneHop(t, false) })
+	t.Run("Remote", func(t *testing.T) { testOneHop(t, true) })
+}
+
+func testOneHop(t *testing.T, remote bool) {
 	defer goleak.VerifyNone(t, goleak.IgnoreCurrent())
 	logrus.SetOutput(ioutil.Discard)
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
 	defer cancel()
-	domain := sandbox.NewBuilder(t).
-		SetNodesCount(2).
-		SetRegistryProxySupplier(nil).
-		SetContext(ctx).
-		Build()
-	defer domain.Cleanup()
 
-	nseReg := &registry.NetworkServiceEndpoint{
-		Name:                "final-endpoint",
-		NetworkServiceNames: []string{"my-service-remote"},
+	var nodesCount, nscNode, expectedPathSegments int
+	if remote {
+		nodesCount = 2
+		nscNode = 1
+		expectedPathSegments = 8
+	} else {
+		nodesCount = 1
+		nscNode = 0
+		expectedPathSegments = 5
 	}
```

Comment:
Please consider extracting testSample struct  and using _for_ loop over samples

```go
type testSample struct {
  nodeCount uint
  nscNode uint
  nseNode uint
  reRequestCount unit
  expectedPathSegments unit
  expectedreRequestCount unit 
}
```

Reply:
I don't feel that the only two cases worth generalizing.

Reply:
1. Actually you already generalized these two cases
2. With `testSample` we can cover many cases such as 
    2.1. nsc - node1, nse - node 2
    2.2. nsc - node2, nse - node1
    2.3. nsc - node1, nse -node1 (nodesCount=1),
    2.4. nsc  - node2, nse -node2 (nodesCount=2)
    2.6. all above scenarious with reRequestsCount = 2
    2.7. all above scenarious with reRequestsCount = 5
    2.7. all above scenarious with reRequestsCount = 10
3. test pattern with testSample looks more better than adding method with bolean logic.